torch/csrc/jit/runtime/symbolic_shape_registry.cpp

```diff
@@ -131,6 +131,62 @@ const std::string shape_compute_functions =
               out.append(li[i])
           return out
 
+        def index_select(self: List[int], dim: int, index: List[int]):
+          dim = maybe_wrap_dim(dim, len(self))
+          numel = multiply_integers(index)
+          assert len(index) <= 1
```

Comment:
jw why pass in `List[int]` if it has to be basically an int? Is this necessitated by some public facing api somewhere?

Reply:
The operator signature is `aten::index_select(Tensor self, int dim, Tensor index)`, and here this is logic that is being replicated here

https://codebrowser.bddppq.com/pytorch/pytorch/aten/src/ATen/native/TensorAdvancedIndexing.cpp.html#858 -